Automattic/wp-calypso
# Framework: Remove unused lib/promises

Since #12975 removed all of the menus code, some libs and utils are no longer used. One of them is `lib/promises` and its `makePromiseSequence`. This PR deletes it.

To test:
* Checkout this branch
* Verify `makePromiseSequence` is indeed no longer used
